core/src/main/java/org/elasticsearch/common/lucene/index/ElasticsearchDirectoryReader.java

```diff
@@ -76,4 +74,38 @@ public LeafReader wrap(LeafReader reader) {
         }
     }
 
+    @SuppressForbidden(reason = "This is the only sane way to add a ReaderClosedListener")
+    public static void addReaderCloseListener(IndexReader reader, IndexReader.ReaderClosedListener listener) {
```

Comment:
It would be good in the future if we could refine this to just DirectoryReader? Really IMO nobody should be installing a `readerCloseListener` on a LeafReader... caching etc should always be managed via `coreCloseListener`